rawhttp-core/src/main/java/rawhttp/core/RequestLine.java

```diff
@@ -85,12 +85,18 @@ public String toString() {
         }
         try {
             // only path and query are sent to the server
-            pathURI = new URI(null, null, null, -1, path, uri.getQuery(), null);
```

Comment:
Wouldn't it have worked if you only had changed `uri.getQuery()` to `uri.getRawQuery()`?

Reply:
I had a look and it seems that it would've been impossible to use the query like this, because if the original query in `uri` contained a escaped character, it would be un-escaped here by calling `getQuery()`, but then the original query could never be recovered... this is a problem with URL encoding: it's one-way only. Once you decode it, you can't know the original unless you keep each part separated and encode them individually. This is why `pathURI` could never include the original query below.
Thanks for picking this up @jmini  I will merge and clean up this a little bit as the use of a new `URI` in this code becomes redundant now.

Reply:
I also thought `uri.getRawQuery()` was the solution… But as you wrote it is not the case.